### PR TITLE
feat: [CDS-91008]: Write schema for K8sApply, K8sScale, and K8sRollingRollback steps

### PR DESCRIPTION
**Pull request by shalini-agr**

### CODE REVIEW

File: v1/pipeline/steps/cd/k8s-apply-step-info.yaml [hunk 1]
```diff
@@ -3,99 +3,45 @@ allOf:
...
-      - type: string
-        pattern: ^<\+input>((\.)((executionInput\(\))|(allowedValues|default|regex)\(.+?\)))*$
-        minLength: 1
+        - type: array
+          items:
+            type: string
+          maxItems: 256
+          minItems: 1
+        - type: string
+          pattern: ^<\+input>((\.)((executionInput\(\))|(allowedValues|default|regex)\(.+?\)))*$
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/k8s-apply-step-info.yaml:28]: This pattern is not valid for V1.

Comment (shalini-agr) [reply v1/pipeline/steps/cd/k8s-apply-step-info.yaml]: removed the pattern

File: v1/pipeline/steps/cd/k8s-apply-step-info.yaml [hunk 2]
```diff
@@ -3,99 +3,45 @@ allOf:
...
+          minLength: 1
     overrides:
       type: array
       items:
         $ref: manifest-config-wrapper.yaml
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/k8s-apply-step-info.yaml:29]: @shalini-agr this `manifest-config-wrapper.yaml` later references the `manifest-config.yaml` and it has V0 schema structure like below. So lets update that as well.
```
required:
  - identifier
  - spec
  - type
 ```

File: v1/pipeline/steps/cd/k8s-rolling-rollback-step-info.yaml [hunk 3]
```diff
@@ -3,48 +3,18 @@ allOf:
...
-        pattern: ^<\+input>((\.)((executionInput\(\))|(allowedValues|default|regex)\(.+?\)))*$
-        minLength: 1
-    delegateSelectors:
+        - type: array
+          items:
+            $ref: k8s-step-command-flag.yaml
+        - type: string
+    delegates:
+      $ref: ../../common/delegate.yaml
+    pruning:
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/k8s-rolling-rollback-step-info.yaml:15]: In k8s-rolling-deploy step we are calling this `prune`.
File name: `k8s-rolling-step-info.yaml`

File: v1/pipeline/steps/cd/k8s-scale-step-info.yaml [hunk 4]
```diff
@@ -3,48 +3,18 @@ allOf:
 - $ref: ../../common/step-spec-type.yaml
 - type: object
   required:
-  - instanceSelection
-  - workload
+    - instanceSelection
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/k8s-scale-step-info.yaml:6]: should follow snake case for field name instanceSelection -> instance_selection

File: v1/pipeline/steps/cd/k8s-scale-step-info.yaml [hunk 5]
```diff
@@ -3,48 +3,18 @@ allOf:
...
-      - type: string
-        pattern: (<\+.+>.*)
-        minLength: 1
-    instanceSelection:
+    delegates:
+      $ref: ../../common/delegate.yaml
+    instance_selection:
       $ref: instance-selection-wrapper.yaml
-    skipSteadyStateCheck:
+    skip_steady_state_check:
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/k8s-scale-step-info.yaml:13]: At one place we are following the skip: then internal keys that we want to skip. But here we are flattening the skip_keys.
Same for `skip_dry_run` in k8s-rolling-step-info.yaml
And at another place its 
skip:
  dry_run: 
  steady_state_check: 

We should follow one pattern.

File: v1/pipeline/steps/cd/k8s-rolling-rollback-step-info.yaml (hunk 3 above)
Comment (shalini-agr) [reply v1/pipeline/steps/cd/k8s-rolling-rollback-step-info.yaml]: done

File: v1/pipeline/steps/cd/k8s-scale-step-info.yaml (hunk 5 above)
Comment (shalini-agr) [reply v1/pipeline/steps/cd/k8s-scale-step-info.yaml]: done

File: v1/pipeline/steps/cd/k8s-scale-step-info.yaml (hunk 4 above)
Comment (shalini-agr) [reply v1/pipeline/steps/cd/k8s-scale-step-info.yaml]: done

File: v1/pipeline/steps/cd/git-lab-store.yaml [hunk 6]
```diff
@@ -3,22 +3,23 @@ allOf:
...
   required:
-  - connectorRef
-  - gitFetchType
+  - connector
+  - fetch_type
   properties:
     branch:
       type: string
-    commitId:
+    commit_id:
```
Comment (brijeshdhakar) [inline v1/pipeline/steps/cd/git-lab-store.yaml:11]: Shall we call `commit_id` -> `commit` ?

Comment (shalini-agr) [reply v1/pipeline/steps/cd/git-lab-store.yaml]: done

**Review by brijeshdhakar — APPROVED** [review]

**Review by archit-harness — APPROVED** [review]

**Comment by bot2-harness** [comment]
This pull request has been automatically closed due to 30 days of inactivity. If you believe this was done in error, please feel free to reopen the PR or reach out to the BT team for further assistance. Thank you for your contributions!

**Pull request closed without merging**